### PR TITLE
Ignore acknowledge for task status update, received out of order.

### PR DESCRIPTION
**Pull request by varungup90**
Ignore acknowledge for task status update, received out of order.

WIP: Integration test.

### CODE REVIEW

File: src/slave/task_status_update_manager.cpp [hunk 1]
```diff
@@ -881,11 +881,21 @@ void TaskStatusUpdateStream::_handle(
     // Add it to the pending updates queue.
     pending.push(update);
   } else {
-    // Record this ACK.
-    acknowledged.insert(id::UUID::fromBytes(update.uuid()).get());
+    StatusUpdate& pendingUpdate = pending.front();
```
Comment (zhitaoli) [inline src/slave/task_status_update_manager.cpp]: Make this const &

File: src/slave/task_status_update_manager.cpp [hunk 2]
```diff
@@ -881,11 +881,21 @@ void TaskStatusUpdateStream::_handle(
...
 
-    // Remove the corresponding update from the pending queue.
-    pending.pop();
+    if (pendingUpdate.uuid() == update.uuid()) {
```
Comment (zhitaoli) [inline src/slave/task_status_update_manager.cpp]: Can we use the _operator==()_ on the _id::UUID_ to compare the two fields instead?

**Comment by bmahler** [comment]
Can you include a precise description of the bug?

**Comment by greggomann** [comment]
@vargup do you know if this is still an issue? Is there an associated JIRA?

**Comment by cf-natali** [comment]
Closing for staleness and lack of activity.

**Pull request closed without merging**